src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,62 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.EmptyFieldException;
+import core.basesyntax.exception.IllegalAgeException;
+import core.basesyntax.exception.InvalidLoginException;
+import core.basesyntax.exception.InvalidPasswordException;
+import core.basesyntax.exception.LoginExistingException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final String EMPTY_FIELD_MESSAGE = "Login/password or age is empty. "
+            + "Please, write correct input data to continue!";
+    private static final String INVALID_LOGIN_MESSAGE = "Login %s is incorrect. "
+            + "Your login cannot be less than 6 characters!";
+    private static final String INVALID_PASSWORD_MESSAGE = "Password %s is incorrect. "
+            + "Your password cannot be less than 6 characters!";
+    private static final String INCORRECT_AGE_MESSAGE = "Age %d is incorrect. "
+            + "Age cannot be less than 18 years old or over the %d!";
+    private static final String LOGIN_EXISTS_MESSAGE = "Login %s is already in storage. "
+            + "Please, write another login to continue!";
+    private static final int MAX_PERMISSIBLE_AGE = 122;
```

Comment:
[Common mistakes](https://github.com/proxychang/jv-registration-validation/blob/hw-jv-registration-validation/checklist.md#common-mistakes-jv-registration-validation)
- Make sure to not add something that is not described in the task requirements, or tied to something that may change over time like max user age. It is likely that average and maximum human lifespan will grow over time, so let's keep it in mind.